### ISSUE TITLE
Support restarting the rogue server and keep the same pysmurf client session running or not?

### ISSUE THREAD

**Issue by swh76** [slaclab/pysmurf#476]
## Describe the problem

We haven't made a deliberate choice on this but currently, pysmurf is written so that restarting the rogue server and keeping the same pysmurf client session running is supported (ie nearly every register is re-polled before it's used, even if it's essentially a static register in rogue).  We could reduce register polls if we cached some of these things (like `SmurfApplication:SmurfVersion`).

**Comment by jesusvasquez333** [comment]
@swh76 I don't think you should cache the `SmurfVersion` in this case, as if you want to support restarting the rogue server, them the new server could be potentially a different version.

**Comment by jesusvasquez333** [comment]
Do you have to reinstantiate the `pysmurf.client.SmurfControl` object once the server is restarted @swh76 ? if that is the case, then you can read the version there, and then consider it don't change, until the next time the object is instantiated.

**Comment by ruck314** [comment]
Looking back at this issue: I believe it can be closed. The design question ("cache static registers vs. support rogue server restart") was settled by code evolution rather than by an explicit fix:

1. **Performance concern moved out of pysmurf.** PR #799 (rogue v6 migration, commit 43e6b999) replaced the EPICS `caget`/`caput` path with pyrogue's `VirtualClient`. The pysmurf-level PV cache that existed in the EPICS era (commits 256d6efb, ddb9dc3c) was removed; pyrogue now does its own internal caching in C++. `_caget` in `smurf_command.py` no longer caches anything at the pysmurf layer (`self._pv_cache` is a vestigial empty dict).

2. **Restart support is explicit.** Open PR #908 (issues-538) hardens `get_pysmurf_version()` callers against `None` returns in `set_defaults_pv`, `set_check_jesd`, and `get_jesd_status` — exactly the "rogue is restarting / unavailable" failure mode @jesusvasquez333 flagged above.

3. **Users who want to read versions once now have a clean API.** Open PR #922 (issues-492) adds `SmurfUtilMixin.get_versions()`, which aggregates pysmurf / rogue / FPGA / startup / cryocard / system version info into a single dict — the right way to "snapshot" versions without introducing a cache that would silently lie after a server restart.

Net effect: pysmurf supports rogue server restart with the same client session, and the per-call polling cost is now absorbed by pyrogue's own caching layer rather than by a parallel cache in pysmurf. Recommending closure once #908 and #922 land.
